src/components/time/__internal__/time-toggle/time-toggle.component.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useRef, useCallback } from "react";
+import guid from "../../../../__internal__/utils/helpers/guid";
+import { TagProps } from "../../../../__internal__/utils/helpers/tags";
+import useLocale from "../../../../hooks/__internal__/useLocale";
+
+import ButtonToggleGroup from "./time-toggle.style";
+import { ButtonToggle } from "../../../button-toggle";
+
+export type ToggleValue = "AM" | "PM";
+export interface ToggleDataProps {
+  wrapperProps?: Omit<TagProps, "data-component">;
+  amToggleProps?: Omit<TagProps, "data-component">;
+  pmToggleProps?: Omit<TagProps, "data-component">;
+}
+
+export interface ToggleProps extends Omit<TagProps, "data-component"> {
+  size?: "small" | "medium" | "large";
+  onChange: (pressedValue: ToggleValue) => void;
+  toggleValue: ToggleValue;
+  disabled?: boolean;
+  toggleProps?: ToggleDataProps;
+}
+
+const Toggle = ({
+  size,
+  onChange,
+  toggleValue,
+  disabled,
+  toggleProps,
+}: ToggleProps) => {
+  const locale = useLocale();
+  const amText = locale.time.amText();
+  const pmText = locale.time.pmText();
+  const internalId = useRef(guid());
+  const { wrapperProps, amToggleProps, pmToggleProps } =
+    toggleProps || /* istanbul ignore next */ {};
+
+  const handleChange = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
question(non-blocking): Just for my own curiosity really. I think we've used `ChangeEvent` in other change handlers. Why is this `MouseEvent`?

Reply:
That's what the type is in the [ButtonToggleGroup interface](https://github.com/Sage/carbon/blob/9d7460f72d77e1eb1c2d7ea02b20b022aa5187f1/src/components/button-toggle/button-toggle-group/button-toggle-group.component.tsx#L55)